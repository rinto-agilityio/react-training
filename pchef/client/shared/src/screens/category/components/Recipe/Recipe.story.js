// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

// Components
import Recipe from '.'

// Mock
import { recipes } from '../../../../mocks'

// Helper
import { wInfo } from '../../../../../.storybook/utils';


storiesOf('Recipe in category', module)
  .addDecorator(wInfo())
  .add('Default', () => (
    <Recipe
      recipe={recipes[0]}
      size="small"
      handlePressImage={action('press image')}
      handlePressIcon={action('press icon')}
    />
  ))
