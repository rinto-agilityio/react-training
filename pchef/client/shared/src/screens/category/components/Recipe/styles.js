// Lib
import { StyleSheet, Dimensions } from 'react-native'

// Thmemes
import { METRICS, FONTS, COLORS } from '../../../../themes'

// Define screen width
const screenWidth = Math.round(Dimensions.get('window').width)

// Create style for Recipe
const styles = StyleSheet.create({
  recipe: {
    // width: screenWidth,
    width: '33.33%',
    borderBottomWidth: METRICS.smallBorderWidth,
    borderColor: COLORS.baseGray,
    marginLeft: METRICS.mediumMargin,
    marginRight: METRICS.mediumMargin,
    marginBottom: METRICS.largeMargin,
  },
  wrapper: {
    height: METRICS.smallWrapper,
    width: screenWidth - METRICS.mediumImage - METRICS.largeMargin,
    marginLeft: METRICS.mediumMargin,
    marginRight: METRICS.mediumMargin,
    marginBottom: METRICS.mediumMargin,
  },
  wrapperText: {
    width: '100%',
    overflow: 'hidden',
    height: 130,
  },
  smallText: {
    fontSize: FONTS.fontSize.base,
  },
  image: {
    width: METRICS.mediumImage,
    height: METRICS.mediumImage,
  },
  icon: {
    position: 'absolute',
    bottom: METRICS.smallPosition,
    right: METRICS.mediumPosition,
  },
  largeContent: {
    // width: METRICS.extraLargeScreen,
    height: METRICS.smallContent,
    position: 'relative',
  },
  largeWrapper: {
    width: METRICS.extraLargeScreen - METRICS.largeImage - METRICS.largeMargin,
    height: METRICS.largeWrapper,
    paddingLeft: METRICS.mediumPadding,
  },
  largeImage: {
    width: METRICS.largeImage,
    height: METRICS.largeImage,
  },
  title: {
    fontSize: FONTS.fontSize.large,
    paddingBottom: METRICS.mediumPadding,
  },
  largeTitle: {
    fontSize: FONTS.fontSize.extraLarge,
  },
  largeText: {
    fontSize: FONTS.fontSize.medium,
    lineHeight: METRICS.mediumLineHeight,
  },
})

export default styles
