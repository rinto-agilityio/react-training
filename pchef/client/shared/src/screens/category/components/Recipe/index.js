// @flow
// add flow above to fix for using flow with React.memo

// Libs
import React, { memo } from 'react'
import { View, Text } from 'react-native'

// Commons
import Wrapper from '../../../../layout/Wrapper'
import Icon from '../../../../components/Icon'
import Image from '../../../../components/Image'

// Constants and Helpers
import { formatStringToArray } from '../../../../helpers/utils'
import { SEPARATOR_SPLIT_STRING, ELLIPSIS_LINE, DEFAULT_IMAGE } from '../../../../constants/index'

// Themes
import { COLORS, FONTS } from '../../../../themes'

// Styles
import styles from './styles'

type Props = {
  recipe: {
    id: string,
    title: string,
    imgUrl: string,
    description: string,
  },
  isGrid: boolean,
  isFavorite: boolean,
  onPressIcon?: (id: string) => void | Promise<void>,
  size?: string,
  color?: string,
  handlePressImage?: () => void,
  customIconStyle: Object,
}

const Recipe = ({
  recipe,
  onPressIcon = () => {},
  size = '',
  color,
  handlePressImage,
  isGrid,
  isFavorite,
  customIconStyle,
}: Props) => {
  const { id, title, imgUrl, description } = recipe
  const formatDescription =
    formatStringToArray(description, SEPARATOR_SPLIT_STRING) || []

  return (
    <View style={[styles.recipe, isGrid ? styles[`${size}GridContent`] : styles[`${size}ListContent`]]}>
      <Text
        numberOfLines={ELLIPSIS_LINE}
        ellipsizeMode="tail"
        style={[styles.title, styles[`${size}Title`]]}
      > {title}
      </Text>
      <Wrapper direction="row" childPosition="left">
        <Image
          url={imgUrl || DEFAULT_IMAGE}
          customImageStyle={[styles.image, styles[`${size}Image`]]}
          handleTouch={handlePressImage}
        />
        <Wrapper
          direction="column"
          childPosition="middle"
          customStyles={[styles.wrapper, isGrid ? styles[`${size}GridWrapper`] : styles[`${size}Wrapper`]]}
        >
          {/* ingredient description */}
          <View style={styles.wrapperText}>
            {formatDescription.map((item, index) => (
              <Text
                numberOfLines={ELLIPSIS_LINE}
                ellipsizeMode="tail"
                key={index}
                style={styles[`${size}Text`]}
              >
                {item}
              </Text>
            ))}
          </View>

          <Icon
            wrapperIconStyle={[styles.icon, customIconStyle]}
            name="favorite"
            size={
              size === 'large'
                ? FONTS.fontSize.extraLarge
                : FONTS.fontSize.medium
            }
            onPress={() => onPressIcon(id)}
            color={isFavorite ? COLORS.red : color}
          />
        </Wrapper>
      </Wrapper>
    </View>
  )
}

Recipe.defaultProps = {
  onPressIcon: () => {},
  size: 'medium',
  color: COLORS.grayDarker,
  handlePressImage: () => {},
}

export default memo<Props>(Recipe)
