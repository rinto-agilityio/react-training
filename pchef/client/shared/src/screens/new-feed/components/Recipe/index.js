// @flow
// add flow above to fix for using flow with React.memo

// Libs
import React, { memo } from 'react'
import { Text, View } from 'react-native'

// Styles
import styles from './styles'

// Components
import Image from '../../../../components/Image'
import Reaction from '../../../../components/Reaction'

// Helpers
import {
  checkContainField,
  formatFiledOnObject,
  checkContain,
} from '../../../../helpers/utils'

// Constants
import { DEFAULT_IMAGE } from '../../../../constants'

import type { RecipeType } from '../../../../types'

type Props = {
  recipe: RecipeType,
  size?: string,
  favoriteRecipe: Array<{
    id: string,
  }>,
  userToggleRecipe: (
    recipeId: string,
    favoriteRecipe: Array<{ id: string }>
  ) => Promise<{ data: { userToggleRecipe: { results: Array<string> } } }>,
  handleClickRecipe: (recipeId: string) => void,
  userToggleVote: (
    recipeId: string,
    votes: Array<string>,
    userId: string
  ) => Promise<{ data: { userToggleVote: { results: Array<string>}}}>,
  userId: string,
  wrapperIconStyle: Object,
}

const Recipe = ({
  recipe,
  size = 'large',
  favoriteRecipe,
  userToggleRecipe,
  handleClickRecipe,
  userToggleVote,
  userId,
  wrapperIconStyle,
}: Props) => {
  const { id, title, description, thumbnail, votes } = recipe

  const isFavorited = checkContainField(favoriteRecipe, id)

  const handleSaveRecipe = async () => {
    await userToggleRecipe(id, favoriteRecipe).then(({ data }) => {
      const {
        userToggleRecipe: { results },
      } = data

      if (results) {
        checkContainField(formatFiledOnObject(results), id)
      }
    })
  }

  const handleToggleVote = async () => {
    await userToggleVote(id, votes, userId)
      .then(({ data }) => {
        const {
          userToggleVote: { results },
        } = data

        if (results) {
          checkContain(votes, id)
        }
      })
  }

  return (
    <View style={[styles.wrapper, styles[`${size}Wrapper`]]}>
      <View style={styles.recipe}>
        <Text
          style={[styles.title, styles[`${size}Title`]]}
          numberOfLines={1}
          ellipsizeMode="tail"
          onPress={() => handleClickRecipe(recipe.id)}
        >
          {title}
        </Text>
        <Text
          style={[styles.description, styles[`${size}Description`]]}
          numberOfLines={1}
          ellipsizeMode="tail"
        >
          {description}
        </Text>
        <Image
          url={thumbnail || DEFAULT_IMAGE}
          customImageStyle={[styles.image, styles[`${size}Image`]]}
          handleTouch={() => handleClickRecipe(recipe.id)}
        />
      </View>
      <Reaction
        votes={votes}
        size={size}
        isFavorited={isFavorited}
        onPressFavorite={handleSaveRecipe}
        onPressVote={handleToggleVote}
        isVote={checkContain(votes, userId)}
        wrapperIconStyle={wrapperIconStyle}
      />
    </View>
  )
}

Recipe.defaultProps = {
  size: 'large',
}

export default memo<Props>(Recipe)
