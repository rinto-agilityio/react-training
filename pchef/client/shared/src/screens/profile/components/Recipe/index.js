// Libs
import React from 'react'
import { Text, View, TouchableOpacity } from 'react-native'

// Styles
import styles from './styles'

// Components
import Image from '../../../../components/Image'
import Reaction from '../../../../components/Reaction'

// Helpers
import { checkContainField, checkContain } from '../../../../helpers/utils'

// Constanst
import { DEFAULT_IMAGE } from '../../../../constants'

type Props = {
  recipe: {
    id: string,
    title: string,
    description: string,
    imgUrl: string,
    votes: Array<string>,
  },
  size?: string,
  favoriteRecipe: Array<{
    id: string,
  }>,
  userToggleRecipe: (
    recipeId: string,
    favoriteRecipe: Array<{
      id: string,
    }>
  ) => Promise<{ data: { userToggleRecipe: { results: Array<string> } } }>,
  userToggleVote: (
    recipeId: string,
    votes: Array<string>,
    userId: string
  ) => Promise<{ data: { userToggleVote: { results: Array<string>}}}>,
  userId: string,
}

const Recipe = ({
  recipe,
  size = 'large',
  favoriteRecipe,
  userToggleRecipe,
  userToggleVote,
  userId,
}: Props) => {
  const { id, title, description, imgUrl, votes } = recipe

  // Call mutation for save recipe
  const handleSaveRecipe = () => {
    userToggleRecipe(id, favoriteRecipe)
  }

  const handleToggleVote = async () => {
    await userToggleVote(id, votes, userId)
      .then(({ data }) => {
        const {
          userToggleVote: { results },
        } = data

        if (results) {
          checkContain(votes, id)
        }
      })
  }

  const isFavorited = checkContainField(favoriteRecipe, id)

  return (
    <TouchableOpacity style={[styles.wrapper, styles[`${size}Wrapper`]]}>
      <View style={styles.recipe}>
        <Text
          style={[styles.title, styles[`${size}Title`]]}
          numberOfLines={1}
          ellipsizeMode="tail"
        >
          {title}
        </Text>
        <Text
          style={[styles.description, styles[`${size}Description`]]}
          numberOfLines={1}
          ellipsizeMode="tail"
        >
          {description}
        </Text>
        <Image
          url={imgUrl || DEFAULT_IMAGE}
          customImageStyle={[styles.image, styles[`${size}Image`]]}
        />
      </View>
      <Reaction
        votes={votes}
        size={size}
        isFavorited={isFavorited}
        isVote={checkContain(votes, userId)}
        onPressFavorite={handleSaveRecipe}
        onPressVote={handleToggleVote}
      />
    </TouchableOpacity>
  )
}

Recipe.defaultProps = {
  size: 'large',
}

export default Recipe
