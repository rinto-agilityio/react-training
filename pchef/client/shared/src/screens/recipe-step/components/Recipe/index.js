// Libs
import React, { useEffect, useState } from 'react'
import { Text, View } from 'react-native'

// Styles
import styles from './styles'

// Themes
import { COLORS, METRICS } from '../../../../themes'

// Components
import Progress from './ProgressStep'
import ImageBackground from '../../../../components/ImageBackground'
import Reaction from '../../../../components/Reaction'
import Comment from '../../../recipe/components/Comment'
import Loading from '../../../../components/Loading'
import Error from '../../../../components/Error'

// mock data
import { recipes } from '../../../../mocks'

// utils
import { findStep, compareStep, customError } from '../../../../helpers/utils'

type Props = {
  recipeSteps: Array<{
    description: string,
    imgUrl: string,
    step: number,
    title: string,
  }>,
  recipe: {
    title: string,
    votes: Array<number>,
    subTitle: string,
    views: number,
    userId: number,
    steps: Array<{
      step: number
    }>,
  },
  size: string,
  onPress?: () => void,
  customRecipe?: {},
  customTitle?: {},
  customDescription?: {},
  customImage?: {},
  customSubTitle?: {},
  customTitleStep?: {},
  loading: boolean,
  error: string
}

const Recipe = ({
  size = 'large',
  onPress,
  customRecipe,
  customTitle,
  customDescription,
  customImage = {},
  customSubTitle,
  customTitleStep,
  recipeSteps = [{
    description: '',
    imgUrl: '',
    step: 1,
    title: '',
  }],
  loading,
  error,
}: Props) => {
  // order recipeSteps by step asc
  const orderRecipeSteps = recipeSteps.sort(compareStep)
  const defaultStepInfo = orderRecipeSteps[0]
  const [stepInfo, setStepInfo] = useState()

  useEffect(() => (
    setStepInfo(defaultStepInfo)
  ), [loading, defaultStepInfo])

  const {
    title,
    subTitle,
    votes,
    userId,
    views,
  } = recipes[0]

  if (loading) {
    return <Loading />
  }

  if (error) {
    return <Error message={customError(error.graphQLErrors)} />
  }
  /**
   * Handle when user click prev or next icon
   * param {name}
   */
  const onPressSelectStep = name => {
    let nextStepInfo

    switch (name) {
      case 'next':
        nextStepInfo = findStep(orderRecipeSteps, stepInfo.step + 1)
        break;
      case 'prev':
        nextStepInfo = findStep(orderRecipeSteps, stepInfo.step - 1)
        break;
      default:
        break;
    }
    setStepInfo(nextStepInfo)
  }

  /**
   * Handle When user click any step
   * @param {step}
   */
  const onPressStep = step => {
    const stepInfoSelect = findStep(orderRecipeSteps, step)
    setStepInfo(stepInfoSelect)
  }

  return (
    <View style={[styles.wrapper, styles[`${size}Wrapper`]]}>
      <View
        style={[
          styles.recipe,
          customRecipe,
        ]}
      >
        <Text
          style={[
            styles.title,
            styles[`${size}Title`],
            customTitle,
          ]}
          onPress={onPress}
        >
          {title}
        </Text>
        <Text
          style={[
            styles.title,
            styles[`${size}Description`],
            customSubTitle,
          ]}
          onPress={onPress}
        >
          {subTitle}
        </Text>
        <ImageBackground
          url={stepInfo.imgUrl}
          customImageBg={[
            styles.image,
            styles[`${size}Image`],
            customImage,
          ]}
          resizeMode="cover"
        >
          <Text>
            <Text
              style={[
                styles.titleStep,
                styles[`${size}Description`],
                customTitleStep,
              ]}
            >
              {stepInfo.title}
            </Text>
          </Text>
          <Progress
            steps={orderRecipeSteps}
            size={size}
            step={stepInfo.step}
            onPressSelectStep={onPressSelectStep}
            onPressStep={onPressStep}
          />
        </ImageBackground>
      </View>
      <Text
        style={[
          styles.description,
          styles[`${size}Description`],
          customDescription,
        ]}
      >
        {stepInfo.description}
      </Text>
      <Reaction
        votes={votes}
        size={size}
        isFavorited={false}
      />
      <Comment
        name={`by ${userId}`}
        publishedAt={Date.now()}
        isGetTime
        customStyle={{
          paddingLeft: METRICS.largePadding,
        }}
        customNameStyles={{
          fontWeight: 'bold',
        }}
        customContentStyles={{
          fontSize: 12,
          color: COLORS.baseGray,
        }}
      />
      <Text
        style={[
          styles.views,
          styles[`${size}Views`],
        ]}
      >
        {`viewed by ${views}`}
      </Text>
    </View>
  )
}

Recipe.defaultProps = {
  onPress: () => {},
  customRecipe: {},
  customTitle: {},
  customDescription: {},
  customSubTitle: {},
  customTitleStep: {},
  customImage: {},
}

export default Recipe
