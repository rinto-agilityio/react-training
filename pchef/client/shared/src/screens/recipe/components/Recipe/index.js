// Libs
import React, { useState } from 'react'
import { View } from 'react-native'

// Styles
import styles from './styles'

// Components
import Ingredients from './Ingredients'
import Directions from './Directions'
import Loading from '../../../../components/Loading'
import Modal from '../../../../components/Modal'
import Error from '../../components/Error'

// Helper
import { customError } from '../../../../helpers/utils'

type Props = {
  getRecipe: {
    description: string,
    id: string,
  },
  size: string,
  onSelectStep: (id: string) => void,
  loading: boolean,
  error: Object,
  recipeSteps: Array<{
    step: number,
    title: string
  }>,
  history: Object,
}

const Recipe = ({
  getRecipe = {
    description: '',
    id: '',
  },
  size = 'medium',
  onSelectStep,
  loading,
  error,
  recipeSteps = [{
    step: 1,
    title: '',
  }],
  history,
}: Props) => {
  const [visible, setVisible] = useState(true)

  if (loading) {
    return <Loading />
  }

  const handleNavigateLogin = () => {
    setVisible(false)
    history.push('/login')
  }

  if (error) {
    return (
      <Modal
        visible={visible}
        onDismiss={() => handleNavigateLogin()}
        onSubmit={() => handleNavigateLogin()}
        size={size}
      >
        <Error message={customError(error.graphQLErrors)} size="medium" />
      </Modal>
    )
  }

  const {
    description,
    id,
  } = getRecipe

  return (
    <View style={[styles.wrapper, styles[`${size}Wrapper`]]}>
      <Ingredients
        description={description}
        size={size}
      />
      <Directions
        steps={recipeSteps}
        size={size}
        onSelectStep={() => onSelectStep(id)}
      />
    </View>
  )
}

export default Recipe
