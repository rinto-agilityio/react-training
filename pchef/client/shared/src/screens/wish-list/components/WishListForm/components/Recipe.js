// Libs
import React from 'react'
import { Text, View } from 'react-native'

// Styles
import styles from '../styles'

// Components
import Image from '../../../../../components/Image'

// Constants
import { DEFAULT_IMAGE } from '../../../../../constants'

type Props = {
  recipe: {
    title: string,
    imgUrl: string,
  },
  size?: string,  
}

const Recipe = ({
  recipe,
  size = 'large',
}: Props) => {
  const { title, imgUrl } = recipe

  return (
    <View style={[styles.wrapper, styles[`${size}Wrapper`]]}>
      <View style={styles.recipe}>
        <Text
          style={[styles.title, styles[`${size}Title`]]}
          numberOfLines={1}
          ellipsizeMode="tail"
        >
          { title }
        </Text>
        <Image
          url={imgUrl || DEFAULT_IMAGE}
          customImageStyle={[styles.image, styles[`${size}Image`]]}
        />
      </View>
    </View>
  )
}

export default Recipe
