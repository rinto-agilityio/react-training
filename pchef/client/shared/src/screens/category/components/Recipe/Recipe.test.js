// Libs
import React from 'react'
import renderer from 'react-test-renderer'
import { JSDOM } from 'jsdom'

// Components
import Recipe from '.'
import Image from '../../../../components/Image'
import Icon from '../../../../components/Icon'

// Mocks
import { recipes } from '../../../../mocks'

const { document } = new JSDOM('').window
global.document = document
global.window = document.defaultView
global.Image = window.Image

const props = {
  recipe: recipes[0],
}

describe('Components', () => {
  describe('Recipe', () => {
    let wrapper
    let handlePressImage
    let onPressIcon

    beforeEach(() => {
      handlePressImage = jest.fn()
      handlePressImage.mockReturnValue('Image on press invoked')

      onPressIcon = jest.fn()
      onPressIcon.mockReturnValue('Icon on press invoked')

      wrapper = shallow(
        <Recipe
          {...props}
          handlePressImage={handlePressImage}
          onPressIcon={onPressIcon}
        />
      )
    })

    it('should render Header component with grid', () => {
      const renderComponent = renderer.create(<Recipe {...props} />)
      expect(renderComponent).toMatchSnapshot()
    })

    it('should call handlePressImage when press on image', () => {
      wrapper
        .find(Image)
        .props()
        .handleTouch()
      expect(handlePressImage.mock.calls.length).toBe(1)
    })

    it('should call onPressIcon when press on Icon', () => {
      wrapper
        .find(Icon)
        .props()
        .onPress()
      expect(onPressIcon.mock.calls.length).toBe(1)
    })

    it('size icon equal medium if pass medium size to props', () => {
      wrapper.setProps({ size: 'medium' })
      const sizeIcon = wrapper.find(Icon).props().size

      expect(sizeIcon).toEqual(18)
    })

    it('should have default onPressIcon props', () => {
      Recipe.defaultProps.onPressIcon()
      expect(Recipe.defaultProps.onPressIcon).toBeDefined()
    })

    it('should have default handlePressImage props', () => {
      Recipe.defaultProps.handlePressImage()
      expect(Recipe.defaultProps.handlePressImage).toBeDefined()
    })
  })
})
