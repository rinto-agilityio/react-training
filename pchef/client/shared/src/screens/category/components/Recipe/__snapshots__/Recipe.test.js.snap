// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Recipe should render Header component with grid 1`] = `
<div
  className="css-view-1dbjc4n r-borderBottomWidth-qklmqi r-marginBottom-117bsoe r-marginLeft-1n0xq6e r-marginRight-zso239"
>
  <div
    className="css-text-76zvg2 r-fontSize-adyw6z r-paddingBottom-1mi0q7o"
    dir="auto"
  >
    Bun Bo Hue Recipe
  </div>
  <div
    className="css-view-1dbjc4n r-alignItems-1habvwh r-display-6koalj r-flexDirection-18u37iz r-flexWrap-1w6e6rj r-justifyContent-1h0z5md"
    style={
      Object {
        "WebkitBoxFlex": 0,
        "WebkitFlexGrow": 0,
        "flexGrow": 0,
        "msFlexPositive": 0,
      }
    }
  >
    <div
      className="css-view-1dbjc4n r-cursor-1loqt21 r-touchAction-1otgn73 r-transitionDuration-eafdt9 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
      data-focusable={true}
      disabled={false}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      tabIndex="0"
    >
      <div
        className="css-view-1dbjc4n r-flexBasis-1mlwlqe r-height-7r4507 r-overflow-1udh08x r-zIndex-417010"
        style={
          Object {
            "width": "150px",
          }
        }
      >
        <div
          className="css-view-1dbjc4n r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-backgroundSize-4gszlv r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw"
          style={
            Object {
              "backgroundImage": "url(\\"https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg\\")",
            }
          }
        />
        <img
          alt=""
          className="css-accessibilityImage-9pa8cd"
          draggable={false}
          src="https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg"
        />
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-alignItems-1habvwh r-display-6koalj r-flexDirection-eqz5dr r-flexWrap-1w6e6rj r-height-7r4507 r-justifyContent-1wtj0ep r-marginBottom-15d164r r-marginLeft-1n0xq6e r-marginRight-zso239"
      style={
        Object {
          "WebkitBoxFlex": 0,
          "WebkitFlexGrow": 0,
          "flexGrow": 0,
          "msFlexPositive": 0,
        }
      }
    >
      <div
        className="css-text-76zvg2 r-fontSize-ubezar r-height-9njtsq r-overflow-1udh08x"
        dir="auto"
      >
        2lb pork neck bones, 2lb beef bones, 1lb oxtail
      </div>
      <div
        className="css-view-1dbjc4n r-alignItems-1awozwy r-bottom-1ebgqk7 r-flexDirection-18u37iz r-position-u8s1d r-right-r19jdo"
      >
        <div
          className="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-1niwhzg r-borderRadius-6ncur5 r-cursor-1loqt21 r-height-1r8g8re r-justifyContent-1777fci r-margin-5scogr r-overflow-bztko3 r-touchAction-1otgn73 r-userSelect-lrvibr r-width-1acpoxo"
          data-focusable={true}
          disabled={false}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="css-hitSlop-mjp8i1"
            style={
              Object {
                "bottom": "-6px",
                "left": "-6px",
                "right": "-6px",
                "top": "-6px",
              }
            }
          />
          <div
            className="css-view-1dbjc4n"
          >
            <div
              className="css-text-76zvg2 r-backgroundColor-1niwhzg r-pointerEvents-633pao"
              dir="auto"
              focusable={false}
              role="img"
              style={
                Object {
                  "color": "rgba(128,128,128,1.00)",
                  "fontSize": "18px",
                }
              }
            >
              â–¡
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
